src/main/java/com/multithreading/TextProcessor.java

```diff
@@ -0,0 +1,38 @@
+package com.multithreading;
+
+class Capitalize implements Runnable {
+    private final char[] chars;
+    private final int index;
+
+    public Capitalize(char[] chars, int index) {
+        this.chars = chars;
+        this.index = index;
+    }
+
+    @Override
+    public void run() {
+        synchronized (chars) {
+            chars[index] = Character.toUpperCase(chars[index]);
+        }
+    }
+}
+
+public class TextProcessor {
+
+    public String process(String input) throws InterruptedException {
+        Thread[] threads = new Thread[input.length()];
+
+        char[] chars = input.toCharArray();
+
+        for (int i = 0; i < chars.length; i++) {
```

Comment:
I would have expected to see:
Breakdown into single chars
Run capitalized on every char
**collect the outputs of each thread as it completes**
build the output.

Reply:
not sure if I interpreted your expectations correctly but I gave it another go - the main function will collect the results from the threads after join is called